# Workflow: 

- Step 1.  For 'Position_clean' column, find all rows that contain substring combo ('intern' OR 'co-op) AND ('data science' OR 'data scientist' OR 'ML' or 'machine learning') *case insensitive*
    - replace all with 'Data Scientist Intern' 
- Step 2.  find all rows that contain substring 'data science'*case insensitive*
    - replace case by case with 'xx Data Scientist' 
- Step 3. find all rows that contain substring 'analytics' *case insensitive*
    - replace case by case with 'xx Data Scientist' 
- Step 4. find all rows that contain substring 'ML' or 'machine learning' (assume: exclude 'AI' roles) *case-insensitive*
    - replace case by case with 'xx Data Scientist' 
    - e.g. For Canada: 
            - "Research Scientist, Core ML" -> "Senior Data Scientist" 
            - 'Senior Machine Learning Scientist" -> "Senior Data Scientist" 
            - "Senior Machine Learning Scientist, Natural Language Processing" -> "Senior Data Scientist" 
            - "Sr ML/AI Data Analyst" (4 positions) -> "Data Scientist"
 
- Step 5. Other special cases (from examining df.Position. unique()): replace case by case with 'xx Data Scientist' 
  - For Canada: 
       -'Programmer/Developer Lead - Sr Data Analyst - Visualization' -> "Senior Data Scientist"
                    
- Step 6: find all rows that do NOT contain substring *case-insensitive* 'Data Scientist'(e.g. 'software engineer', 'AI research scientist', 'data engineer', 'software developer', 'database developer', 'data analyst') -> replace with np.nan (for easier removal of these rows later) 


[see here for str.replace](https://stackoverflow.com/questions/39768547/replace-whole-string-if-it-contains-substring-in-pandas)

**note: order matters! e.g. '*Senior* *director*, data scientist' - need to change to LDS, before SR**

- Step 7A. replace any substrings 'lead' or 'director' or 'manager' or 'principal' -> 'Lead Data Scientist'
- Step 7B. replace any substrings 'sr' or 'senior' -> 'Senior Data Scientist'
- Step 7C. replace any substrings jr' or 'junior' -> 'Junior Data Scientist'
- Step 7D. replace substrings NOT 'intern' NOT 'junior' NOT 'senior' NOT 'lead' NOT NaN  -> 'Data Scientist' 


AFTER merging with Sasha's cleaned columns, do following: 
- drop rows that have NaN in the 'Position_clean' column [see how here](https://stackoverflow.com/questions/13413590/how-to-drop-rows-of-pandas-dataframe-whose-value-in-a-certain-column-is-nan)
- check and drop duplicates
- drop other rows with certain NaN fields

In [1]:
import pandas as pd
import numpy as np

In [2]:
INTERN = 'Data Scientist Intern'
JR = 'Junior Data Scientist'
SR = 'Senior Data Scientist'
DS = 'Data Scientist'
LDS = 'Lead Data Scientist'
rm = np.nan

In [3]:
df_us = pd.read_csv('elisa_us_new.csv', index_col = 0)

In [4]:
df_us.Position.unique() 

array(['Jr. Data Scientist', 'Data Scientist-Junior', 'Data Analyst',
       'Scientist III, Pharma R & D', 'Database Developer',
       'Senior Scientist I', 'Energy Research and Data Analyst',
       'Senior, Measurement Data Analyst', 'Associate Data Analyst',
       'Principal Quality Scientist', 'Senior Database Developer',
       'Data Scientist', 'Population Health Data Analyst',
       'Care Coordination Quality Improvement Data Analyst',
       'Web Data Analyst (South of Milwaukee)', 'Enterprise Data Analyst',
       'Financial Data Analyst', 'Contract Data Analyst',
       'Senior Scientist in Product Development', 'Junior Data Scientist',
       'Senior Data Scientist Advisor', 'Senior SQL Database Developer',
       'Sales Data Analyst (Outdoor/Hunting Industry)',
       'Operations Data Analyst', 'DEI Data Analyst',
       'Finance Data Analyst Internship - Paid (South of Milwaukee)',
       'SQL Data Analyst', 'Energy Market Data Analyst',
       'Healthcare Data Analyst

In [5]:
# duplicate
df_us['Position_clean'] = df_us.loc[:, 'Position']

In [6]:
df_us[df_us['Position'] == 'Lead Account Data Scientist']

Position  \
148  Lead Account Data Scientist   

                                           Description          Company  \
148  Expectations\n\n\nHard skills:\n2-3 years in a...  Bell Integrator   

         Location Glassdoor Salary Estimate Rating                    Size  \
148  San Jose, CA                        NF    3.4  1001 to 5000 Employees   

                      Revenue               Type Year Founded     Industry  \
148  Unknown / Non-Applicable  Company - Private         2003  IT Services   

                     Sector               Position_clean  
148  Information Technology  Lead Account Data Scientist

In [7]:
## step 1
s = ['intern', 'co-op', 'co op', 'placem']
cond_or1 = '|'.join(s)
t = ["data scien", "machine learn"]
cond_or2 = '|'.join(t)
bool_intern = (df_us['Position'].str.contains(cond_or1, case = False))& \
               ( (df_us['Position'].str.contains(cond_or2, case = False)) |\
                (df_us['Position'].str.contains('ML', case = True))
               ) 
df_us[bool_intern] # display rows

Position  \
53                              Data Scientist, Intern   
75                               Data Scientist Intern   
81                               Data Scientist Intern   
89                               Data Scientist Intern   
97                              Data Scientist, Intern   
118                              Data Scientist Intern   
123                              Data Scientist Intern   
132                              Data Scientist Intern   
140                              Data Scientist Intern   
145                              Data Scientist Intern   
165                              Data Scientist Intern   
179                              Data Scientist Intern   
182                              Data Scientist Intern   
187                              Data Scientist Intern   
196                              Data Scientist Intern   
198                              Data Scientist Intern   
2                                Data Scientist Intern   
40                               Data Scientist Intern   
102                              Data Scientist Intern   
175   Data Scientist Intern - Strategic Data Solutions   
63                     Research Data Scientist, Intern   
88   2021 Campus: Data Scientist Intern (Walmart Ec...   
98                               Data Scientist Intern   
196                  Intern - Photomask Data Scientist   
22                        Data Scientist Intern - UIUC   
35   2021 Campus: Data Scientist Intern (Walmart Ec...   
48                               Data Scientist Intern   
79                               Data Scientist Intern   
109                              Data Scientist Intern   
87                                 Data Science Intern   

                                           Description  \
53   Help us shape the future of education.\n\nOur ...   
75   Intern - Data Science\n\nThe New York Times is...   
81   Internet Brands is looking for a Data Scientis...   
89   Joby Overview: Located in Northern California,...   
97   Help us shape the future of education.\n\nOur ...   
118  Worker Sub-Type:\n\nStudent (Fixed Term)\n\nJo...   
123  Bayer is a global enterprise with core compete...   
132  ** Elutions will be unable to sponsor or trans...   
140  Build the Future\n\nDo you enjoy testing the l...   
145  HENKEL IS FOR THOSE WHO STEP UP. DO YOU?\nAt H...   
165  Intern - Data Science\n\nThe New York Times is...   
179  Description\n\nJob Description:\nLeidos is see...   
182  HENKEL IS FOR THOSE WHO STEP UP. DO YOU?\n\n\n...   
187  Description\n\nJob Description:\nLeidos is see...   
196  Job ID: JR0145964\n\nJob Category: Intern/Stud...   
198  Internet Brands is looking for a Data Scientis...   
2    Come Join Analog Devices – a place where Innov...   
40   Joby Overview: Located in Northern California,...   
102  AMPLY is a fast-growing 2020 Global CleanTech ...   
175  Posted: Nov 6, 2020\nWeekly Hours: 40\nRole Nu...   
63   Facebook's mission is to give people the power...   
88   Position Summary...\nWhat you'll do...\nApplie...   
98   Internet Brands is looking for a Data Scientis...   
196  Relocation Level: Domestic\nHiring Manager: Jo...   
22   YOUR TASKS AND RESPONSIBILITIES\n\nThe primary...   
35   Position Summary...\nWhat you'll do...\nApplie...   
48   AMPLY is a fast-growing 2020 Global CleanTech ...   
79   Internet Brands is looking for a Data Scientis...   
109  Internet Brands is looking for a Data Scientis...   
87   The Opportunity\n\n\nWe are seeking a talented...   

                      Company           Location Glassdoor Salary Estimate  \
53                   Duolingo     Pittsburgh, PA                        NF   
75         The New York Times       New York, NY                        NF   
81            Internet Brands    Los Angeles, CA                        NF   
89              Joby Aviation     San Carlos, CA                        NF   
97                   Duolingo     Pittsburgh, PA          

In [8]:
df_us.loc[bool_intern, 'Position_clean'] = INTERN
# 1c. check by comparing 2 columns
df_us[bool_intern][['Position', 'Position_clean']]

Position         Position_clean
53                              Data Scientist, Intern  Data Scientist Intern
75                               Data Scientist Intern  Data Scientist Intern
81                               Data Scientist Intern  Data Scientist Intern
89                               Data Scientist Intern  Data Scientist Intern
97                              Data Scientist, Intern  Data Scientist Intern
118                              Data Scientist Intern  Data Scientist Intern
123                              Data Scientist Intern  Data Scientist Intern
132                              Data Scientist Intern  Data Scientist Intern
140                              Data Scientist Intern  Data Scientist Intern
145                              Data Scientist Intern  Data Scientist Intern
165                              Data Scientist Intern  Data Scientist Intern
179                              Data Scientist Intern  Data Scientist Intern
182                              Data Scientist Intern  Data Scientist Intern
187                              Data Scientist Intern  Data Scientist Intern
196                              Data Scientist Intern  Data Scientist Intern
198                              Data Scientist Intern  Data Scientist Intern
2                                Data Scientist Intern  Data Scientist Intern
40                               Data Scientist Intern  Data Scientist Intern
102                              Data Scientist Intern  Data Scientist Intern
175   Data Scientist Intern - Strategic Data Solutions  Data Scientist Intern
63                     Research Data Scientist, Intern  Data Scientist Intern
88   2021 Campus: Data Scientist Intern (Walmart Ec...  Data Scientist Intern
98                               Data Scientist Intern  Data Scientist Intern
196                  Intern - Photomask Data Scientist  Data Scientist Intern
22                        Data Scientist Intern - UIUC  Data Scientist Intern
35   2021 Campus: Data Scientist Intern (Walmart Ec...  Data Scientist Intern
48                               Data Scientist Intern  Data Scientist Intern
79                               Data Scientist Intern  Data Scientist Intern
109                              Data Scientist Intern  Data Scientist Intern
87                                 Data Science Intern  Data Scientist Intern

In [9]:
## Step 2
bool_2 = df_us['Position'].str.contains('data science', case = False)
df_us[bool_2] # display

Position  \
136                 Data Analyst I - Data Science Team   
137  Junior Data Scientist Data Science Development...   
145                                   VP, Data Science   
173                                   VP, Data Science   
208  Translational Data Scientist, Oncology Data Sc...   
11                    Data Science & Analytics Manager   
98         Senior Manager of Data Science - Operations   
107  Translational Data Scientist, Oncology Data Sc...   
135                   Data Science & Analytics Manager   
89                  Senior/Staff Data Science Engineer   
101                          Manager, Data Science TPM   
8                                Manager, Data Science   
52                        Senior Manager, Data Science   
87                  Senior/Staff Data Science Engineer   
159                          Manager, Data Science TPM   
179  Senior Manager, Online Data Science - Recommen...   
10   Senior Manager, Online Data Science - Recommen...   
73                  Senior/Staff Data Science Engineer   
12                                Data Science Manager   
87                             Data Science Fellowship   
162                   Software Engineer - Data Science   
176        Data Science Instructor, Part-Time - REMOTE   
177                               Data Science Manager   
178                Research Scientist II, Data Science   
24   Manager, Data Science - Data Analytics (Phoeni...   
29   Translational Data Scientist, Oncology Data Sc...   
87                                 Data Science Intern   
139  Manager, Data Science - Data Analytics (Phoeni...   
148                             Director, Data Science   
155  Translational Data Scientist, Oncology Data Sc...   
161  Translational Data Scientist, Oncology Data Sc...   
1                             Director of Data Science   
21                        Data Science Support Analyst   
42                                Data Science Manager   
47                     Head of Real-World Data Science   
73   Associate Director, Data Science - Lexington, ...   
76                    Data Science & Analytics Manager   
90                        Data Science Support Analyst   
132                               Data Science Manager   
32   Associate Director, Data Science - Lexington, ...   
39                             Sr Data Science Analyst   
45                     Head of Real-World Data Science   
78              Data Science Engineer, Recent Graduate   
146        Senior Manager of Data Science - Operations   
2                              Sr Data Science Analyst   
20                    Data Science & Analytics Manager   
35                    Data Science & Analytics Manager   
71              Data Science Engineer, Recent Graduate   
103                                       Data Science   

                                           Description  \
136  Wednesday, October 14, 2020\n\nThe Data Scienc...   
137  Do you want to be part of a team that encourag...   
145  Oversee a team of Data Scientists with overall...   
173  Oversee a team of Data Scientists with overall...   
208  18-Sep-2020\n\nJob ID\n303124BR\n{"QuestionNam...   
11   If you ever wanted to be on the ground floor o...   
98   Guild is hiring a Senior Manager of Data Scien...   
107  18-Sep-2020\n\nJob ID\n303124BR\n{"QuestionNam...   
135  If you ever wanted to be on the ground floor o...   
89   Are you an experience data engineer/ scientist...   
101  Why This Role is Important to Us:\nThe Manager...   
8    For best results, use Google Chrome to view th...   
52   Atlassian is continuing to hire with all inter...   
87   Are you an experience data engineer/ scientist...   
159  Why This Role is Important to Us:\nThe Manager...   
179  Position Description:\nPosition Purpose:\n\nAt...   
10   Position Description:\nPosition Purpose:\n\nAt...   
73   Are you an experience data engineer/ scientist...   
12   Come join a growing financial technolo

In [10]:
# special case 1 
df_us.loc[df_us['Position'] == 'Data Science Support Analyst', 'Position_clean'] = rm
df_us.loc[df_us['Position'] == 'Data Science Instructor, Part-Time - REMOTE', 'Position_clean'] = rm

In [11]:
# special case 2
df_us.loc[df_us['Position'] == 'Data Science', 'Position_clean'] = DS
# check 
#df_us[df_us['Position'] == 'Data Science']

In [12]:
# jr ds
bool_jr = (df_us['Position_clean'].str.contains('junior|jr|recent grad|Analyst I', case = False)) & \
           (df_us['Position_clean'].str.contains('data science', case = False))
df_us.loc[bool_jr, 'Position_clean'] = JR
# lead ds
bool_lds = (df_us['Position_clean'].str.contains('VP|head|manager|director|lead|principal', case = False)) & \
           (df_us['Position_clean'].str.contains('data science', case = False))
df_us.loc[bool_lds, 'Position_clean'] = LDS
# sr ds
bool_sr = (df_us['Position_clean'].str.contains('senior/staff|fellowship|senior|sr', case = False)) & \
           (df_us['Position_clean'].str.contains('data science', case = False))
df_us.loc[bool_sr, 'Position_clean'] = SR
# ds
bool_ds = (df_us['Position_clean'].str.contains('software engineer|Scientist II', case = False)) & \
           (df_us['Position_clean'].str.contains('data science', case = False))
df_us.loc[bool_ds, 'Position_clean'] = DS

In [13]:
# check 
df_us[bool_2][['Position', 'Position_clean']]

Position  \
136                 Data Analyst I - Data Science Team   
137  Junior Data Scientist Data Science Development...   
145                                   VP, Data Science   
173                                   VP, Data Science   
208  Translational Data Scientist, Oncology Data Sc...   
11                    Data Science & Analytics Manager   
98         Senior Manager of Data Science - Operations   
107  Translational Data Scientist, Oncology Data Sc...   
135                   Data Science & Analytics Manager   
89                  Senior/Staff Data Science Engineer   
101                          Manager, Data Science TPM   
8                                Manager, Data Science   
52                        Senior Manager, Data Science   
87                  Senior/Staff Data Science Engineer   
159                          Manager, Data Science TPM   
179  Senior Manager, Online Data Science - Recommen...   
10   Senior Manager, Online Data Science - Recommen...   
73                  Senior/Staff Data Science Engineer   
12                                Data Science Manager   
87                             Data Science Fellowship   
162                   Software Engineer - Data Science   
176        Data Science Instructor, Part-Time - REMOTE   
177                               Data Science Manager   
178                Research Scientist II, Data Science   
24   Manager, Data Science - Data Analytics (Phoeni...   
29   Translational Data Scientist, Oncology Data Sc...   
87                                 Data Science Intern   
139  Manager, Data Science - Data Analytics (Phoeni...   
148                             Director, Data Science   
155  Translational Data Scientist, Oncology Data Sc...   
161  Translational Data Scientist, Oncology Data Sc...   
1                             Director of Data Science   
21                        Data Science Support Analyst   
42                                Data Science Manager   
47                     Head of Real-World Data Science   
73   Associate Director, Data Science - Lexington, ...   
76                    Data Science & Analytics Manager   
90                        Data Science Support Analyst   
132                               Data Science Manager   
32   Associate Director, Data Science - Lexington, ...   
39                             Sr Data Science Analyst   
45                     Head of Real-World Data Science   
78              Data Science Engineer, Recent Graduate   
146        Senior Manager of Data Science - Operations   
2                              Sr Data Science Analyst   
20                    Data Science & Analytics Manager   
35                    Data Science & Analytics Manager   
71              Data Science Engineer, Recent Graduate   
103                                       Data Science   

                                        Position_clean  
136                              Junior Data Scientist  
137                              Junior Data Scientist  
145                                Lead Data Scientist  
173                                Lead Data Scientist  
208  Translational Data Scientist, Oncology Data Sc...  
11                                 Lead Data Scientist  
98                                 Lead Data Scientist  
107  Translational Data Scientist, Oncology Data Sc...  
135                                Lead Data Scientist  
89                               Senior Data Scientist  
101                                Lead Data Scientist  
8                                  Lead Data Scientist  
52                                 Lead Data Scientist  
87                               Senior Data Scientist  
159                                Lead Data Scientist  
179                                Lead Data Scientist  
10                                 Lead Data Scientist  
73                               Senior Data Scientist  
12                                 Lead Data Scientist  
87   

In [14]:
## step 3: 'analytics'
bool_3 = df_us['Position'].str.contains('analytics', case = False)
#df_us[bool_3]['Position'].iloc[0:] # display

In [15]:
# special 1: remove analytics engineer
bool_eng = df_us['Position_clean'].str.contains('analytics', case = False, na = False) &\
           df_us['Position_clean'].str.contains('engineer', case = False, na = False)
df_us.loc[bool_eng, 'Position_clean'] = rm
# special 2: remove irrelevant
df_us.loc[df_us['Position'] == 'Senior Product Manager - Threat Detection and Analytics - Secureworks - Remote, US', 'Position_clean'] = rm
df_us.loc[df_us['Position'] == 'Sign Out Manager-Data Analytics & Results Virtual Hiring Event' , 'Position_clean'] = rm
df_us.loc[df_us['Position'] == 'Data Analyst, Performance Analytics', 'Position_clean'] = rm
# special 3
df_us.loc[df_us['Position'] == 'Scientist 2, Data Analytics', 'Position_clean'] = DS
df_us.loc[df_us['Position'] == 'Aon Campus - Health Analytics Data Analyst', 'Position_clean'] = DS

In [16]:
# jr ds
bool_jr = (df_us['Position_clean'].str.contains('junior|jr', case = False, na = False)) & \
           (df_us['Position_clean'].str.contains('analytics', case = False, na = False))
df_us.loc[bool_jr, 'Position_clean'] = JR
# lead ds
bool_lds = (df_us['Position_clean'].str.contains('VP|head|manag|director|lead|principal', case = False, na = False)) & \
           (df_us['Position_clean'].str.contains('analytics', case = False, na = False))
df_us.loc[bool_lds, 'Position_clean'] = LDS
# sr ds
bool_sr = (df_us['Position_clean'].str.contains('senior|sr|advance', case = False, na = False)) & \
           (df_us['Position_clean'].str.contains('analytics', case = False, na = False))
df_us.loc[bool_sr, 'Position_clean'] = SR
# ds
bool_ds = df_us['Position_clean'].str.contains('analytics', case = False, na = False)
df_us.loc[bool_ds, 'Position_clean'] = DS

In [17]:
# check 
df_us[bool_3][['Position', 'Position_clean']].iloc[40:60,]

Position         Position_clean
156  CIB Wholesale Payments Data and Analytics - Da...    Lead Data Scientist
158                    Lead Data Scientist - Analytics    Lead Data Scientist
160          Data Scientist, Analytics-Remote Presence         Data Scientist
178            Enterprise Analytics Data Scientist Sr.  Senior Data Scientist
18                              Risk Analytics Manager    Lead Data Scientist
42   Sign Out Manager-Data Analytics & Results Virt...                    NaN
80                          Business Analytics Manager    Lead Data Scientist
105   Corporate Analytics Manager (South of Milwaukee)    Lead Data Scientist
112                Data Analyst, Performance Analytics                    NaN
163                 Data Scientist, Business Analytics         Data Scientist
189                        Marketing Analytics Manager    Lead Data Scientist
195                        Marketing Analytics Manager    Lead Data Scientist
17                             Data Analytics Engineer                    NaN
24   Manager, Data Science - Data Analytics (Phoeni...    Lead Data Scientist
33   Data Scientist, Product Analytics - Epidemic R...         Data Scientist
71   Customer Service Analytics Manager (South of M...    Lead Data Scientist
74     Sr. Manager Advanced Analytics (Data Scientist)    Lead Data Scientist
78                             Analytics Data Engineer                    NaN
128       Senior Data Scientist, Analytics & Inference  Senior Data Scientist
139  Manager, Data Science - Data Analytics (Phoeni...    Lead Data Scientist

In [18]:
## step 4 - omit since all relevant ML roles also have 'data scientist' in their titles so can deal w it in Step 7
## Step 5  - other special cases
# spec 1
bool_s = df_us['Position_clean'].str.contains('statistician|data mining|python', case = False) &\
        df_us['Position_clean'].str.contains('data analyst', case = False)
#df_us[bool_s]
df_us.loc[bool_s, 'Position_clean'] = JR
# spec 2
df_us.loc[df_us['Position'] == 'Data & Applied Scientist', 'Position_clean'] = DS
df_us.loc[df_us['Position'] == 'Sr. Data & Applied Scientist', 'Position_clean'] = SR
df_us.loc[df_us['Position'] == 'Quantitative Analyst and Risk Modeler', 'Position_clean'] = DS

In [19]:
## Step 6 - replace all non-DS with NaN
bool_notDS = df_us['Position_clean'].str.contains('Data Scientist', case = False) == False
df_us.loc[bool_notDS, 'Position_clean'] = rm
# check 
# a = df_us[df_us['Position'].str.contains('Data Scientist', case = False) == False]
# a[['Position', 'Position_clean']].shape 

In [20]:
# check no. of unique data scientist listings left after removing irrelevant rows
df_us[df_us['Position_clean'].str.contains('Data Scientist', case = False) == True]['Position'].unique().shape

(331,)

In [31]:
df_us['Position'].unique().shape

(793,)

## Out of 793 *unique* U.S. listings, get 331 *unique* data scientist (includes: 8 intern, 29 Jr., 101 Sr., 100 Lead, 93 data scientist) listings

In [21]:
## Step 7 
# 7A
# special
df_us.loc[df_us['Position'] == 'Data Scientist to Data Scientist Lead', 'Position_clean'] = DS
df_us.loc[df_us['Position'] == 'Lead Account Data Scientist', 'Position_clean'] = DS
# replace w LDS
bool_lds = df_us['Position_clean'].str.contains('lead|principal|manage|direct|VP|head', case = False, na = False)
df_us[bool_lds]['Position'].unique().shape # 100 unique LDS roles
df_us.loc[bool_lds, 'Position_clean'] = LDS

In [23]:
# 7B
bool_sr = df_us['Position_clean'].str.contains('senior|sr|expert|staff|st 3|st III|st II|clinical|intermediate|advance', case = False, na = False)
#df_us[bool_sr]['Position'].unique().shape # 101 unqiue SR roles
df_us.loc[bool_sr, 'Position_clean'] = SR
#check 
df_us[bool_sr][['Position', 'Position_clean']]

Position         Position_clean
26                       Senior Data Scientist Advisor  Senior Data Scientist
57                                  Sr. Data Scientist  Senior Data Scientist
61       Senior Engineer/Data Scientist - Industry 4.0  Senior Data Scientist
36                                Staff Data Scientist  Senior Data Scientist
112                   Staff Data Scientist - Interests  Senior Data Scientist
136                            Clinical Data Scientist  Senior Data Scientist
155                            Clinical Data Scientist  Senior Data Scientist
68                    Staff Data Scientist - Interests  Senior Data Scientist
98                    Staff Data Scientist - Interests  Senior Data Scientist
128                   Staff Data Scientist - Interests  Senior Data Scientist
0                                Senior Data Scientist  Senior Data Scientist
1                Senior Data Scientist (Remote Option)  Senior Data Scientist
14                             Sr Data Scientist, Zoro  Senior Data Scientist
15                               Senior Data Scientist  Senior Data Scientist
17                               Senior Data Scientist  Senior Data Scientist
19                               Senior Data Scientist  Senior Data Scientist
23                               Senior Data Scientist  Senior Data Scientist
26        Senior Research Statistician- Data Scientist  Senior Data Scientist
27                               Senior Data Scientist  Senior Data Scientist
30                               Senior Data Scientist  Senior Data Scientist
31      Senior Data Scientist - Defense & Intelligence  Senior Data Scientist
41               Senior Data Scientist, Masters Degree  Senior Data Scientist
42                              Staff Data Scientist I  Senior Data Scientist
45                    Senior Data Scientist Model Risk  Senior Data Scientist
46                               Senior Data Scientist  Senior Data Scientist
49   Senior Data Scientist (Palo Alto, CA or Remote...  Senior Data Scientist
51               Sr. Data Scientist - Model Governance  Senior Data Scientist
54                       Senior Data Scientist Advisor  Senior Data Scientist
56                                  Sr. Data Scientist  Senior Data Scientist
57                                   Sr Data Scientist  Senior Data Scientist
..                                                 ...                    ...
58                              Data Scientist 3 (844)  Senior Data Scientist
64                Senior Data Scientist - Underwriting  Senior Data Scientist
68             Senior Data Scientist (Remote-friendly)  Senior Data Scientist
92                               Senior Data Scientist  Senior Data Scientist
98                               Senior Data Scientist  Senior Data Scientist
109                    Senior Data Scientist Marketing  Senior Data Scientist
123                              Senior Data Scientist  Senior Data Scientist
128       Senior Data Scientist, Analytics & Inference  Senior Data Scientist
129            Staff Software Engineer, Data Scientist  Senior Data Scientist
133                              Senior Data Scientist  Senior Data Scientist
142    Senior Data Scientist 3 Artificial Intelligence  Senior Data Scientist
153                              Senior Data Scientist  Senior Data Scientist
169    Senior Data Scientist 4 Artificial Intelligence  Senior Data Scientist
173                                  Data Scientist II  Senior Data Scientist
174  Senior Data Scientist - Manufacturing and Supp...  Senior Data Scientist
179                                 Sr. Data Scientist  Senior Data Scientist
4                                    Data Scientist II  Senior Data Scientist
17                                  Data Scientist III  Senior Data Scientist
45                               Data Scientist-Senior  Senior Data Scientist
108                              Senior Data Scientist  Senior Data 

In [25]:
# 7C
bool_jr = df_us['Position_clean'].str.contains('junior|jr|early|entry|new grad|st 2|st 1', case = False, na = False) 
#df_us[bool_jr]
df_us[bool_jr]['Position'].unique().shape 
df_us.loc[bool_jr, 'Position_clean'] = JR

In [26]:
# 7D
bool_remain = (df_us['Position_clean'].str.contains('senior', case = False, na = False) == False) &\
                (df_us['Position_clean'].str.contains('junior', case = False, na = False) == False) &\
                (df_us['Position_clean'].str.contains('lead', case = False, na = False) == False) &\
                (df_us['Position_clean'].str.contains('intern', case = False, na = False) == False) &\
                df_us['Position_clean'].notna()
df_us.loc[bool_remain, 'Position_clean'] = DS

In [30]:
# Final step: check 
df_us[['Position', 'Position_clean']].iloc[300:400,]

# check special cases (handwritten * 2)
df_us[df_us['Position'] == 'Senior AI/ML/NLP Data Scientist']

Position  \
108  Senior AI/ML/NLP Data Scientist   
117  Senior AI/ML/NLP Data Scientist   

                                           Description Company    Location  \
108  Ayata is developing its Prescriptive Analytics...   Ayata  Dallas, TX   
117  Ayata is developing its Prescriptive Analytics...   Ayata  Dallas, TX   

    Glassdoor Salary Estimate Rating               Size  \
108                        NF    2.4  1 to 50 Employees   
117                        NF    2.4  1 to 50 Employees   

                        Revenue               Type Year Founded    Industry  \
108  Less than $1 million (USD)  Company - Private           NF  Accounting   
117  Less than $1 million (USD)  Company - Private           NF  Accounting   

                 Sector         Position_clean  
108  Accounting & Legal  Senior Data Scientist  
117  Accounting & Legal  Senior Data Scientist

In [50]:
# check 
df_us['Position_clean'].unique()

array(['Junior Data Scientist', nan, 'Data Scientist',
       'Senior Data Scientist', 'Lead Data Scientist',
       'Data Scientist Intern'], dtype=object)

In [51]:
# write to csv
df_us.to_csv('elisa_position_us.csv')

In [53]:
# read in to check 
df = pd.read_csv('elisa_position_us.csv', index_col = 0)
df.shape

(2717, 13)